virtio-devices/src/vhost_user/blk.rs

```diff
@@ -254,11 +304,14 @@ impl VirtioDevice for Blk {
     ) -> std::result::Result<(), crate::Error> {
         if self.acked_protocol_features & VhostUserProtocolFeatures::CONFIGURE_MEM_SLOTS.bits() != 0
         {
-            add_memory_region(&mut self.vhost_user_blk, region)
+            add_memory_region(&mut self.vhost_user_blk.lock().unwrap(), region)
                 .map_err(crate::Error::VhostUserAddMemoryRegion)
         } else if let Some(guest_memory) = &self.guest_memory {
-            update_mem_table(&mut self.vhost_user_blk, guest_memory.memory().deref())
-                .map_err(crate::Error::VhostUserUpdateMemory)
+            update_mem_table(
+                &mut self.vhost_user_blk.lock().unwrap(),
+                guest_memory.memory().deref(),
+            )
+            .map_err(crate::Error::VhostUserUpdateMemory)
```

Comment:
You forgot to update the `resume()` function so that it unparks the reconnection thread:
```
if let Some(reconnect_epoll_thread) = &self.reconnect_epoll_thread {
    reconnect_epoll_thread.thread().unpark();
}
```

Reply:
Thanks, fixed it.